### PR TITLE
gitpod-cli: call segement directly for analytics

### PR DESCRIPTION
**Pull request by andreafalzetti**
## Description
Removes a dependency on server to send analytics events from `gp` CLI. 

Supervisor is aware of the OwnerId and we expose it in WorkspaceInfo response.

Screenshot 2023-01-17 at 23 05 34

## Related Issue(s)

Fixes #

## How to test
1. Start a workspace https://af-gp-anal1fcb7cb30d.preview.gitpod-dev.com/workspaces
2. Run `gp rebuild` and generate some events
3. Check in Segment Staging Trusted and verify the event is there, including `UserId`

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
- [x] /werft analytics=segment

### CODE REVIEW

File: components/supervisor-api/info.proto [hunk 1]
```diff
@@ -91,4 +91,6 @@ message WorkspaceInfoResponse {
     }
     // workspace_class denotes the class of the workspace
     WorkspaceClass workspace_class = 15;
+
+    string owner_id = 16;
```
Comment (akosyakov) [inline components/supervisor-api/info.proto:95]: Please add a comment here 🙏🏻

File: components/supervisor/pkg/supervisor/supervisor.go [hunk 2]
```diff
@@ -262,6 +262,13 @@ func Run(options ...RunOption) {
 			WorkspaceID:       cfg.WorkspaceID,
 			SupervisorVersion: Version,
 		}, tokenService)
+
+		ownerID, err := gitpodService.GetOwnerID(ctx)
+		if err != nil {
+			log.WithError(err).Error("gitpod perf analytics: failed to resolve workspace info")
+		}
+
+		cfg.OwnerId = ownerID
```
Comment (akosyakov) [inline components/supervisor/pkg/supervisor/supervisor.go:271]: I don’t config should be changed like that, it is loaded and validated in config.go

**Comment by jeanp413** [comment]
@andreafalzetti  Just found that there's `GITPOD_OWNER_ID` env variable so we can use that and parse it like the other config variables,  we can also remove the `GetOwnerID` method in `publicapi.go`

**Comment by andreafalzetti** [comment]
> @andreafalzetti Just found that there's `GITPOD_OWNER_ID` env variable so we can use that and parse it like the other config variables, we can also remove the `GetOwnerID` method in `publicapi.go`

@jeanp413 is that a good idea, tho? I thought we were trying to avoid relying on env vars if we can? cc @akosyakov

File: components/supervisor/pkg/supervisor/services.go [hunk 3]
```diff
@@ -652,6 +652,7 @@ func (is *InfoService) WorkspaceInfo(context.Context, *api.WorkspaceInfoRequest)
 		IdeAlias:             is.cfg.IDEAlias,
 		IdePort:              uint32(is.cfg.IDEPort),
 		WorkspaceClass:       &api.WorkspaceInfoResponse_WorkspaceClass{Id: is.cfg.WorkspaceClass},
+		OwnerId:              os.Getenv("GITPOD_OWNER_ID"),
```
Comment (akosyakov) [inline components/supervisor/pkg/supervisor/services.go:655]: we should do then is.cfg.OwnerID and read it from env vars, as other configs

**Comment by andreafalzetti** [comment]
moving to draft until I clean up commits & test in prev env.

**Comment by andreafalzetti** [comment]
ready for review again cc @akosyakov

**Review by akosyakov — APPROVED** [review]
/hold

@jeanp413 to have a look again

**Comment by andreafalzetti** [comment]
@jeanp413 could you have a look? 🙏

**Review by jeanp413 — APPROVED** [review]
/unhold

**Pull request merged**